### ISSUE TITLE
RFC: Global use of short array syntax

### ISSUE THREAD

**Issue by michalkleiner** [silverstripe/silverstripe-framework#9066]
_Creating the first issue on the framework repo to start the discussion._

Should SilverStripe switch to using short syntax arrays `[]` (as opposed to the legacy `array()` or a mixture)?

The code can be bulk updated with a tool. The bigger part that can't be that easily/automated are code snippets in doc blocks.

Minimal version of PHP does no longer require support for the keyword array syntax.

**Comment by robbieaverill** [comment]
Yep. We'd welcome a pull request to do so, for SilverStripe 4.0 and up.

You could also consider adding a rule to the phpcs ruleset to enforce this, however we don't use a shared file across all modules, so would need to be added individually as each module has its syntax updated.
